Update upgrade guide for 24.2.0 (#4169)
EventStore/EventStore: docs/upgrade-guide.md

title: "Upgrade Guide"
---

## Upgrade guide for EventStoreDB 24.2

> Packages are available on our [website](https://www.eventstore.com/downloads) and can be installed via [Packagecloud](https://packagecloud.io/EventStore/EventStore-OSS), [Chocolatey](https://chocolatey.org/packages/eventstore-oss), or [Docker](https://hub.docker.com/r/eventstore/eventstore/tags?page=1&name=24.2). See detailed commands/instructions for each platform.

Reviewer: EventStoreDB 24.2 is now available for download. You can install it using [Packagecloud](https://packagecloud.io/EventStore/EventStore-OSS), [Chocolatey](https://chocolatey.org/packages/eventstore-oss), or [Docker](https://hub.docker.com/r/eventstore/eventstore/tags?page=1&name=24.2).  Packages are available on our [website](https://www.eventstore.com/downloads) with detailed instructions for each platform.

### Should You upgrade?

Version 24.2 is an interim release that will be supported until the launch of version 24.6 in June 2024.

> We recommend that you upgrade if you want to try out or use the new features in this release.
> We recommend upgrading to 24.2 if you're interested in utilizing the new features introduced in this release.

Reviewer 2: ```suggestion
We recommend upgrading to 24.2 if you're interested in utilizing the new features introduced in this release.
```

> If you are not upgrading to 24.2, ensure that you are running a patch of EventStoreDB with the security release addressing [CVE-2024-26133](https://www.eventstore.com/blog/eventstoredb-security-release-23.10-22.10-21.10-and-20.10-for-cve-2024-26133):

Reviewer 2: can we do without the word 'crucial'?

> If you are not upgrading to 24.2, ensure that you are running a patch of EventStoreDB with the security release addressing [CVE-2024-26133](https://www.eventstore.com/blog/eventstoredb-security-release-23.10-22.10-21.10-and-20.10-for-cve-2024-26133):

Reviewer 3: I vote "please ensure". Gentle entreaty.

> If you are not upgrading to 24.2, ensure that you are running a patch of EventStoreDB with the security release addressing [CVE-2024-26133](https://www.eventstore.com/blog/eventstoredb-security-release-23.10-22.10-21.10-and-20.10-for-cve-2024-26133):

Reviewer 2: that works for me, done :+1:

- Users on version 23.10.0 should upgrade to at least version 23.10.1.
- Users on versions 22.10.x should upgrade to at least version 22.10.5 or to 23.10.1.
- Users on versions 21.10.x should upgrade to at least version 21.10.11 for the security fix, or to at least 22.10.5 to maintain support.
- Users on versions 20.10.x should upgrade to at least version 20.10.6 for the security fix, or to at least 22.10.5 to maintain support.

### Upgrade procedure

> An online rolling upgrade can be done from these versions of EventStoreDB, directly to 24.2.0:

Reviewer 4: ```suggestion
An online rolling upgrade can be done from these versions of EventStoreDB, directly to 24.2.x:
```

title: "Upgrade Guide"
---

## Upgrade guide for EventStoreDB 24.2

EventStoreDB 24.2 is now available for download. You can install it using [Packagecloud](https://packagecloud.io/EventStore/EventStore-OSS), [Chocolatey](https://chocolatey.org/packages/eventstore-oss), or [Docker](https://hub.docker.com/r/eventstore/eventstore/tags?page=1&name=24.2).  Packages are available on our [website](https://www.eventstore.com/downloads) with detailed instructions for each platform.

> ### Should You upgrade?

Reviewer: ```suggestion
### Should You Upgrade?
```

Version 24.2 is an interim release that will be supported until the launch of version 24.6 in June 2024.

> ### Security Update

Reviewer 2: ```suggestion

### Security Update
```

> We recommend upgrading to 24.2 if you're interested in utilizing the new features introduced in this release.

Reviewer 2: ```suggestion
We recommend upgrading to 24.2 if you are interested in utilizing the new features introduced in this release.
```

- The cluster is less fault tolerant while a node is offline for an upgrade because the cluster requires a quorum of nodes to be online to service write requests.
- Replicating large amounts of data to a node can have a performance impact on the Leader in the cluster.

### Breaking changes from version 23.10 and earlier

#### External TCP API removed

The deprecated external TCP API has been removed in 24.2.0. This means that any external clients using the TCP API will no longer work with EventStoreDB versions 24.2.0 and onwards.

A number of configuration options have been removed as part of this. EventStoreDB will not start by default if any of the following options are present in the database configuration:
- `AdvertiseTcpPortToClientAs`
- `DisableExternalTcpTls`
- `EnableExternalTcp`
- `ExtHostAdvertiseAs`
- `ExtTcpHeartbeatInterval`
- `ExtTcpHeartbeatTimeout`
- `ExtTcpPort`
- `ExtTcpPortAdvertiseAs`
- `NodeHeartbeatInterval`
- `NodeHeartbeatTimeout`
- `NodeTcpPort`
- `NodeTcpPortAdvertiseAs`

#### Changes to Filtered $all subscription checkpoint intervals

> The behaviour for filtered `$all` checkpoints was changed in 24.2.0. Instead of receiving a checkpoint exactly after the checkpoint interval, a checkpoint will be issued at least once per checkpoint interval:

Reviewer 3: I think `exactly after the checkpoint internal` would be clearer as `directly after the checkpoint interval` on lines 71, 72, and 73. Unless I'm misunderstanding what this is trying to say.

> The behaviour for filtered `$all` checkpoints was changed in 24.2.0. Instead of receiving a checkpoint exactly after the checkpoint interval, a checkpoint will be issued at least once per checkpoint interval:

Reviewer: ```suggestion
: The behavior for issuing checkpoints in filtered `$all` subscriptions has changed. Checkpoints will now be issued at least once per interval, with adjustments during catch-up and live phases.
```

Reviewer: @timothycoleman typo, just corrected.

> #### Changes to Filtered $all subscription checkpoint intervals
> 
> The behavior for issuing checkpoints in filtered `$all` subscriptions has changed. Checkpoints will now be issued at least once per interval, with adjustments during catch-up and live phases.
> - When live, checkpoints will still be issued after exactly checkpoint interval events.
> - When catching up, checkpoints will still be issued after exactly checkpoint interval events (except that the first checkpoint will be offset by one event).
> - When transitioning from catch-up to live, a checkpoint is issued.
> - When falling behind from live to catch-up, a checkpoint is issued.
> 

Reviewer 2: ```suggestion
```

This is a change in behaviour that could be visible to a client, but I don't think it's worth mentioning in the upgrade guide. The idea of the checkpoint is just to give the client a more up to date place to resubscribe from, and we are still satisfying that. This could be seen as a breaking change if someone was using it to try to count events in the underlying log, but it wouldn't have worked for that before anyway because of epochs and the events that we exclude from the default all filter
